config_src/drivers/nuopc_cap/mom_cap_methods.F90

```diff
@@ -271,6 +305,159 @@ subroutine mom_import(ocean_public, ocean_grid, importState, ice_ocean_boundary,
     if (ChkErr(rc,__LINE__,u_FILE_u)) return
   endif
 
+  !--------------!
+  ! MARBL fields !
+  !--------------!
+
+  ! seaice_dust_flux, nhx_dep, and noy_dep are single fields from the coupler
+  ! atm_fine_dust_flux, atm_coarse_dust_flux, atm_bc_flux, and seaice_bc_flux
+  ! are all sums of multiple fields and will be treated slightly differently
+  ! For those fields, we use do_sum = .true.
+
+  !----
+  ! nhx deposition
+  !----
+  if (associated(ice_ocean_boundary%nhx_dep)) then
+    call state_getimport(importState, 'Faxa_ndep',  &
+        isc, iec, jsc, jec, ice_ocean_boundary%nhx_dep(:,:), &
+        areacor=med2mod_areacor, esmf_ind=1, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+  endif
+
+    !----
+    ! noy deposition
+    !----
+  if (associated(ice_ocean_boundary%noy_dep)) then
+    call state_getimport(importState, 'Faxa_ndep',  &
+        isc, iec, jsc, jec, ice_ocean_boundary%noy_dep(:,:), &
+        areacor=med2mod_areacor, esmf_ind=2, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+  endif
+
+  !----
+  ! atmospheric CO2 concentration
+  ! might not be passed from atmosphere component,
+  ! in which the pointer(s) will not be associated
+  !----
+  if (associated(ice_ocean_boundary%atm_co2_prog)) then
+    call state_getimport(importState, 'Sa_co2prog',  &
+        isc, iec, jsc, jec, ice_ocean_boundary%atm_co2_prog(:,:), rc=rc)
+  if (ChkErr(rc,__LINE__,u_FILE_u)) return
+  endif
+  if (associated(ice_ocean_boundary%atm_co2_diag)) then
+    call state_getimport(importState, 'Sa_co2diag',  &
+        isc, iec, jsc, jec, ice_ocean_boundary%atm_co2_diag(:,:), rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
```

Comment:
This would be cleaner if the the ChkErr function was used as in other locations
```
if (ChkErr(rc,__LINE__,u_FILE_u)) return
```

Reply:
Thanks for catching this, I will keep a record of them and make code cleanness here in the future PR